core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -49,7 +54,7 @@
      * @param filterBuilder The query to wrap in a constant score query
      */
     public ConstantScoreQueryBuilder(QueryBuilder filterBuilder) {
-        this.filterBuilder = filterBuilder;
+        this.filterBuilder = Objects.requireNonNull(filterBuilder);
```

Comment:
I see what you mean. One option to guard against the NPEs you mention below would be to always initialize inner query builders to the EmptyQueryBuilder, but that way we wouldn't know if the user set it correctly or not. What if in addition to requiring non-null in constructor we also check this in validate(), then we would have to 1.) call validate() at the beginning of doXContenten() to always render legal queries 2.) validate() will be implicitely called before/after transport, so all queries should be validated once toQuery() is called. Am I missing something?